planning/obstacle_avoidance_planner/src/node.cpp

```diff
@@ -560,6 +586,38 @@ void ObstacleAvoidancePlanner::publishDebugData(const Header & header) const
 
   time_keeper_ptr_->toc(__func__, "  ");
 }
+
+void ObstacleAvoidancePlanner::transformPathToOdomFrame(Path::SharedPtr path_ptr) const {
+    geometry_msgs::msg::TransformStamped transform_stamped =
+            map_odom_tf_buffer_->lookupTransform("odom", "map", tf2::TimePointZero);
+
+    path_ptr->header.frame_id="odom";
+    path_ptr->header.stamp = this->now();
+
+    for (auto & point: path_ptr->points) {
+        autoware_auto_planning_msgs::msg::PathPoint odom_path_point;
+        tf2::doTransform(point.pose, point.pose, transform_stamped);
+    }
+
+    for (auto & left_bound_point : path_ptr->left_bound) {
+        tf2::doTransform(left_bound_point, left_bound_point, transform_stamped);
+    }
+
+    for (auto & right_bound_point : path_ptr->right_bound) {
+        tf2::doTransform(right_bound_point, right_bound_point, transform_stamped);
+    }
+}
+
+void ObstacleAvoidancePlanner::convertTrajectoryPointsToMessage
+(const std::vector<TrajectoryPoint> &traj_points) const {
+    auto traj_msg_ptr = std::make_unique<Trajectory>();
+    traj_msg_ptr->header.frame_id = "odom";
+    traj_msg_ptr->header.stamp = this->now();
+    for (auto & traj_point : traj_points){
+        traj_msg_ptr->points.push_back(traj_point);
+    }
+    debug_traj_pub_->publish(*traj_msg_ptr);
```

Comment:
Either use `std::move()` with `std::unique_ptr` or just create the object without a smart pointer. Since the rest of this node doesn't use the `std::unique_ptr` pattern, I recommend just creating the message without a pointer wrapper.

Reply:
Agree. I used `Trajectory::SharedPtr` instead. Just to clarify. This new function seems to be unused here and it was used for converting std::vector<TrajectoryPoint> to trajectory message for visualization and debug purpose. So I still leave it here.